docs/hive-metrics.md

```diff
@@ -0,0 +1,108 @@
+`HiveConfig.Spec.ExportMetrics` specifies whether the operator should enable metrics for hive controllers to be extracted for the prometheus instance running on the cluster. When set to true, `openshift.io/cluster-monitoring` label is applied and  the operator deploys ServiceMonitors so that the prometheus instances can extract metrics. The operator also sets up RBAC in the TargetNamespace so that openshift prometheus in the cluster can list/access objects required to pull metrics.
+
+### Optional Metrics
+
+Some metrics pertaining to Hibernation (or PowerState) controller, are not logged by default, since that feature is not supported across all clouds.
+Admins can opt for logging such metrics via `HiveConfig.Spec.MetricsConfig`
+
+### Metrics with Optional Cluster Deployment labels
+
+Not all metrics are logged per cluster deployment name or namespace. Many metrics are logged per cluster-type, and hive allows for admins to define the labels to look for when reporting these metrics.
+
+Opt into them via `HiveConfig.Spec.MetricsConfig.AdditionalClusterDeploymentLabels`, which accepts a map of the label you would want on your metrics, to the label found on ClusterDeployment.
+For example, including `{"ocp_major_version": "hive.openshift.io/version-major"}` will cause affected metrics to include a label key ocp_major_version with the value from the `hive.openshift.io/version-major` ClusterDeployment label -- e.g. "4".
+
+Note: It is upto the cluster admins to be mindful of cardinality and ensure these labels are not too specific, like cluster id, otherwise it can negatively impact your observability system's performance
+
+### List of all Hive metrics
+
+#### ClusterDeployment controller metrics
+These metrics are logged per ClusterDeployment instance. None of these are optional.
+
+|                       Metric Name                        | Optional Label Support |
+|:--------------------------------------------------------:|:----------------------:|
+|   hive_cluster_deployment_install_job_duration_seconds   |           N            |
+|    hive_cluster_deployment_install_job_delay_seconds     |           N            |
+|    hive_cluster_deployment_imageset_job_delay_seconds    |           N            |
+|        hive_cluster_deployment_dns_delay_seconds         |           N            |
+|    hive_cluster_deployment_completed_install_restart     |           Y            |
+|          hive_cluster_deployments_created_total          |           Y            |
+|         hive_cluster_deployments_installed_total         |           Y            |
+|          hive_cluster_deployments_deleted_total          |           Y            |
+| hive_cluster_deployments_provision_failed_terminal_total |           Y            |
+
+#### ClusterProvision controller metrics
+These metrics are logged per ClusterProvision instance. If they have optional label support, ClusterDeployment will apply those labels to ClusterProvision. None of these are optional.
```

Comment:
Similar to above, "These metrics are observed while processing ClusterProvisions."

> ClusterDeployment will apply those labels to ClusterProvision

What does this mean? Do we get the value from the specified label on the CD or the CP? I would have expected the former, as we still have the CD object in that code flow, and we can't/shouldn't rely on the CP having all the same labels as the CD.